OpenRA.Game/Orders/UnitOrderGenerator.cs

```diff
@@ -140,7 +140,12 @@ static UnitOrderResult OrderForUnit(Actor self, Target target, List<Actor> actor
 			if (mi.Button != Game.Settings.Game.MouseButtonPreference.Action)
 				return null;
 
-			if (self.Owner != self.World.LocalPlayer && (self.World.LocalPlayer.Spectating || !self.World.CommandAlliedUnits || !self.Owner.IsAlliedWith(self.World.LocalPlayer)))
+			// TODO: What is the end goal of the spectating check?
+			// Are we trying to always deny spectators, or only deny them when allied control is on, what's the deal?
```

Comment:
Spectators are marked as allies, is what I guess this is for.

Reply:
I guess my question is do we need an AND or OR here? i.e. which of these is correct?
```c#
if (self.World.LocalPlayer.Spectating)
    return null;
if (!self.World.LocalPlayer.CanControlUnitsOf(self.Owner))
    return null;
```
```c#
if (self.World.LocalPlayer.Spectating && !self.World.LocalPlayer.CanControlUnitsOf(self.Owner))
    return null;
```

Reply:
the first one. spectators shall only spectate in any case